### PR TITLE
Fix for #5280

### PR DESCRIPTION
**Pull request by BaronGreenback**
Url / ports contained in PublishedServerUrl startup option is now assumed to be correct.

### CODE REVIEW

File: MediaBrowser.Controller/IServerApplicationHost.cs [hunk 1]
```diff
@@ -52,6 +52,11 @@ public interface IServerApplicationHost : IApplicationHost
         /// <value>The name of the friendly.</value>
         string FriendlyName { get; }
 
+        /// <summary>
+        /// Gets a value indicating whether this instance has a custom published url.
```
Comment (crobibero) [inline MediaBrowser.Controller/IServerApplicationHost.cs:56]:
```suggestion
        /// Gets the configured published server url.
```

File: Emby.Server.Implementations/ApplicationHost.cs [hunk 2]
```diff
@@ -135,6 +135,11 @@ public abstract class ApplicationHost : IServerApplicationHost, IDisposable
 
         public bool CoreStartupHasCompleted { get; private set; }
 
+        /// <summary>
+        /// Gets a value indicating whether this instance has a custom published url.
+        /// </summary>
```
Comment (crobibero) [inline Emby.Server.Implementations/ApplicationHost.cs:140]:
```suggestion
        /// <inheritdoc />
```

**Review by crobibero — APPROVED** [review]

**Review by cvium — APPROVED** [review]

File: Emby.Dlna/Main/DlnaEntryPoint.cs [hunk 3]
```diff
@@ -313,9 +313,12 @@ private void RegisterServerEndpoints()
                 _logger.LogInformation("Registering publisher for {0} on {1}", fullService, address);
 
                 var uri = new UriBuilder(_appHost.GetSmartApiUrl(address.Address) + descriptorUri);
-                // DLNA will only work over http, so we must reset to http:// : {port}
-                uri.Scheme = "http://";
-                uri.Port = _netConfig.HttpServerPortNumber;
+                if (_appHost.PublishedServerUrl == null)
```
Comment (dbaker-rh) [inline Emby.Dlna/Main/DlnaEntryPoint.cs:316]: This looks incomplete.  If my published server URL specifies a port, I don't see how it'll get picked up.

Comment (dbaker-rh) [reply Emby.Dlna/Main/DlnaEntryPoint.cs:316]: Disregard - I was reading the diff, not the code in context...

**Pull request merged**